NTSC filter improvements

Gumball2415 opened the pull request:
Changes in this PR:
- Add NTSC-CRT library from https://github.com/LMP88959/NTSC-CRT
- Add helper thread in NTSC blargg to improve speed
- Refactor composite dot phase calculation

Gumball2415 committed:
Add NTSC-CRT library

from https://github.com/LMP88959/NTSC-CRT

Gumball2415 committed:
Add modifications for NES NTSC composite

thanks NewRisingSun, PinoBatch, and Trirosmos!

Gumball2415 committed:
Update NTSC-CRT library

Gumball2415 committed:
Begin to connect filter library

Gumball2415 committed:
Implement emulator bindings

Gumball2415 committed:
Fix NTSC emulator settings

- Remove variable resolution
- Fix issue with scanlines
- Fix image offsetting
- Improve signal accuracy

Gumball2415 committed:
Fix hue flickering

Gumball2415 committed:
Fix issues

- Refactor dot skip detection
- Avoid accessing null PPU buffer

Gumball2415 committed:
Increase colorburst integration rate

Thanks Emmir!

Gumball2415 committed:
Directly set carrier phase for faster convergence

- Begin to implement border color

Gumball2415 committed:
Implement border color and gray pulse

- Begin to optimize NTSC filter code

Gumball2415 committed:
Precalculate composite voltage values

Gumball2415 committed:
Properly name resource label

Gumball2415 committed:
Optimize NTSC signal encoding

Gumball2415 committed:
Fix resource labels

Gumball2415 committed:
Set artifact phase as member object

Gumball2415 committed:
Add helper thread for blargg NTSC

Gumball2415 committed:
Optimize NTSC-CRT library

Gumball2415 committed:
Optimize filter

Thanks Emmir!
- Precalculate sync and blanking
- Remove non-essential code
- Begin to integrate frame mix toggle

negativeExponent commented on Core/Core.vcxproj.filters:
```diff
@@ -874,9 +874,6 @@
     <ClInclude Include="Mapper433.h">
       <Filter>Nes\Mappers\Unnamed</Filter>
     </ClInclude>
-    <ClInclude Include="Mapper431.h">
```
why this got removed? is it not working or needed or something?

Gumball2415 replied:
oops, think it got mangled in the middle of including the new filter libraries, will fix this immediately

Gumball2415 committed:
Restore Mapper 431

Gumball2415 committed:
Implement library updates

Thanks Emmir!

Gumball2415 committed:
Add binding for frame blending

Gumball2415 committed:
Sync with library implementation

Gumball2415 committed:
Sync with NTSC-CRT v2.0.0

Gumball2415 committed:
Fix linker issue

Gumball2415 committed:
Normalize IRE level output

Gumball2415 committed:
Remove unnecessary code

Gumball2415 committed:
Adjust default contrast value to avoid clipping

Gumball2415 committed:
Increase sync window

Gumball2415 committed:
Fix broken emphasis colors on Bisqwit NTSC

Gumball2415 committed:
Refactor dot crawl phase

Parity with https://github.com/SourMesen/Mesen2/commit/77acdab5840cbd38624ab9ae14c7ff2359e71e88

Gumball2415 committed:
Force 2-phase dot crawl when overclocking is used

Thanks Sour!

Gumball2415 committed:
Sync with NTSC-CRT v2.2.0

Gumball2415 committed:
Update sync window implementation

Gumball2415 committed:
Calibrate decoding matrix to match Bisqwit's...

palette generator

Gumball2415 committed:
Optimize emphasis waveform generation
